File: proyectomds2/frontend/src/vista-registro.js
import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-upload/src/vaadin-upload.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-text-field/src/vaadin-text-field.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';

class VistaRegistro extends PolymerElement {

    static get template() {
        return html`
<style include="shared-styles">
                :host {
                    display: block;
                    height: 100%;
                }
            </style>
<vaadin-vertical-layout style="width: 100%; height: 100%; padding: var(--lumo-space-l);" id="vaadinVerticalLayout" theme="spacing">
 <vaadin-horizontal-layout theme="spacing" id="horizontal_registrarse" style="width: 100%; height: 100%; flex-direction: row; justify-content: flex-start; align-items: flex-start;">
  <vaadin-vertical-layout theme="spacing" style="width: 100%; height: 100%;" id="vaadinVerticalLayout1">
   <label style="align-self: center;" id="label">Registrarse con:</label>
   <vaadin-horizontal-layout theme="spacing" style="align-self: center;" id="zona_terceros">
    <vaadin-button id="boton_google">
      Google 
    </vaadin-button>
    <vaadin-button id="boton_facebook">
      Facebook 
    </vaadin-button>
    <vaadin-button id="boton_apple">
      Apple ID 
    </vaadin-button>
   </vaadin-horizontal-layout>
   <label style="align-self: center;" id="label1">-o-</label>
   <label style="align-self: center;" id="label2">Registrarse con correo electrónico </label>
   <vaadin-text-field error-message="Please enter a value" required invalid="" label="*" style="width: 100%; height: 100%;" placeholder="Correo electronico" id="text_field_correo"></vaadin-text-field>
   <vaadin-text-field error-message="Please enter a value" required invalid="" label="*" style="width: 100%; height: 100%;" placeholder="Confirmar correo electronico" id="text_field_confirmar_correo1"></vaadin-text-field>
   <vaadin-text-field error-message="Please enter a value" required invalid="" label="*" style="width: 100%; height: 100%;" placeholder="Contraseña" id="text_field_contraseña"></vaadin-text-field>
   <vaadin-text-field error-message="Please enter a value" required invalid="" label="*" style="width: 100%; height: 100%;" placeholder="Confirmar contraseña" id="text_field_confirmar_contraseña"></vaadin-text-field>
   <vaadin-text-field error-message="Please enter a value" required invalid="" label="Required" style="width: 100%;" placeholder="Nombre perfil" id="text_field_usuario"></vaadin-text-field>
   <vaadin-button style="align-self: center;" id="boton_registrarse">
     Registrarte 
   </vaadin-button>
   <vaadin-horizontal-layout theme="spacing" style="align-self: center;" id="vaadinHorizontalLayout">
    <label id="label3">¿Ya tienes cuenta?</label>
    <vaadin-button id="boton_iniciar_sesion">
      Iniciar Sesión 
    </vaadin-button>
   </vaadin-horizontal-layout>
  </vaadin-vertical-layout>
  <vaadin-vertical-layout theme="spacing" style="align-self: flex-end; height: 100%; width: 100%; justify-content: center;" id="vaadinVerticalLayout2">
   <label style="align-self: center;" id="label4">Foto de perfil</label>
   <img style="width: 35%; height: 35%; align-self: center;" src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png" id="imagen">
   <vaadin-upload style="align-self: center; flex-grow: 0;" id="cargar_imagen"></vaadin-upload>
  </vaadin-vertical-layout>
 </vaadin-horizontal-layout>
</vaadin-vertical-layout>
`;
    }

    static get is() {
        return 'vista-registro';
    }

    static get properties() {
        return {
            // Declare your properties here.
        };
    }
}

customElements.define(VistaRegistro.is, VistaRegistro);

File: proyectomds2/src/main/java/vistas/VistaRegistro.java
package vistas;

import com.vaadin.flow.templatemodel.TemplateModel;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.upload.Upload;

/**
 * A Designer generated component for the vista-registro template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("vista-registro")
@JsModule("./src/vista-registro.js")
public class VistaRegistro extends PolymerTemplate<VistaRegistro.VistaRegistroModel> {

    @Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;
	@Id("horizontal_registrarse")
	private HorizontalLayout horizontal_registrarse;
	@Id("textField_correo_electronico")
	private TextField textField_correo_electronico;
	@Id("textField_confirmar_correo")
	private TextField textField_confirmar_correo;
	@Id("textField_contrasena")
	private TextField textField_contrasena;
	@Id("textField_confirmar_contrasena")
	private TextField textField_confirmar_contrasena;
	@Id("textField_nombrePerfil")
	private TextField textField_nombrePerfil;
	@Id("label_yaTienesCuenta")
	private Button label_yaTienesCuenta;
	@Id("horizontal_iniciarSesion")
	private HorizontalLayout horizontal_iniciarSesion;
	@Id("label_yaTienesCuenta1")
	private Label label_yaTienesCuenta1;
	@Id("button_iniciarSesion")
	private Button button_iniciarSesion;
	@Id("vaadinVerticalLayout1")
	private Element vaadinVerticalLayout1;
	@Id("label")
	private Label label;
	@Id("zona_terceros")
	private HorizontalLayout zona_terceros;
	@Id("boton_google")
	private Button boton_google;
	@Id("boton_facebook")
	private Button boton_facebook;
	@Id("boton_apple")
	private Button boton_apple;
	@Id("label1")
	private Label label1;
	@Id("label2")
	private Label label2;
	@Id("text_field_correo")
	private TextField text_field_correo;
	@Id("text_field_confirmar_correo1")
	private TextField text_field_confirmar_correo1;
	@Id("text_field_contraseña")
	private TextField text_field_contraseña;
	@Id("text_field_confirmar_contraseña")
	private TextField text_field_confirmar_contraseña;
	@Id("text_field_usuario")
	private TextField text_field_usuario;
	@Id("boton_registrarse")
	private Button boton_registrarse;
	@Id("vaadinHorizontalLayout")
	private HorizontalLayout vaadinHorizontalLayout;
	@Id("label3")
	private Label label3;
	@Id("boton_iniciar_sesion")
	private Button boton_iniciar_sesion;
	@Id("vaadinVerticalLayout2")
	private Element vaadinVerticalLayout2;
	@Id("label4")
	private Label label4;
	@Id("imagen")
	private Image imagen;
	@Id("cargar_imagen")
	private Upload cargar_imagen;

	/**
     * Creates a new VistaRegistro.
     */
    public VistaRegistro() {
        // You can initialise any data required for the connected UI components here.
    }

    /**
     * This model binds properties between VistaRegistro and vista-registro
     */
    public interface VistaRegistroModel extends TemplateModel {
        // Add setters and getters for template properties here.
    }
}
